src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.test.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { render, cleanup, waitForDomChange } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import ReportsTrafficSources from './ReportsTrafficSources';
+import DopplerIntlProvider from '../../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import { AppServicesProvider } from '../../../services/pure-di';
+
+const errorResponse = { success: false, error: new Error('Dummy error') };
+const fullResponse = {
+  success: true,
+  value: [
+    {
+      sourceName: 'Email',
+      quantity: 2000,
+    },
+    {
+      sourceName: 'Social',
+      quantity: 1000,
+    },
+  ],
+};
+
+describe('reports traffic sources', () => {
+  afterEach(cleanup);
+  it('should deal with DataHub failure', async () => {
+    const dataHubClientDouble = {
+      getTrafficSourcesByPeriod: async () => errorResponse,
+    };
+
+    const domainName = 'doppler.test';
+    const dateFrom = new Date('2019-01-01');
+
+    const { container, getByText } = render(
+      <AppServicesProvider
+        forcedServices={{
+          datahubClient: dataHubClientDouble,
+        }}
+      >
+        <DopplerIntlProvider locale="es">
+          <ReportsTrafficSources domainName={domainName} dateFrom={dateFrom} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+    expect(container.querySelector('.loading-box')).toBeInTheDocument();
+    await waitForDomChange();
+    expect(getByText('trafficSources.error'));
+  });
+
+  it('should show the traffic sources', async () => {
+    const dataHubClientDouble = {
+      getTrafficSourcesByPeriod: async () => fullResponse,
+    };
+
+    const domainName = 'doppler.test';
+    const dateFrom = new Date('2019-01-01');
+
+    const { container, getByText } = render(
+      <AppServicesProvider
+        forcedServices={{
+          datahubClient: dataHubClientDouble,
+        }}
+      >
+        <DopplerIntlProvider locale="es">
+          <ReportsTrafficSources domainName={domainName} dateFrom={dateFrom} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+    expect(container.querySelector('.loading-box')).toBeInTheDocument();
+    await waitForDomChange();
+    expect(getByText(fullResponse.value[0].sourceName));
+    expect(getByText(fullResponse.value[1].sourceName));
+    expect(getByText('66.67%'));
```

Comment:
Nice!

Although, `locale` of `DopplerIntlProvider` is `es`, are we paying attention to that?

Maybe you should use the real providers to test it in English and Spanish.

Reply:
It's the same result. Its only different when have four numbers figures or more like `1000` its `1,000` or `1.000` but in this case it's not different